tests/slice_by_index_list_test.py

```diff
@@ -0,0 +1,97 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+# @file
+# @author Simon Heybrock
+import pytest
+import scipp as sc
+
+
+def make_var() -> sc.Variable:
+    return sc.arange('dummy', 12, dtype='int64').fold(dim='dummy',
+                                                      sizes={
+                                                          'xx': 4,
+                                                          'yy': 3
+                                                      })
+
+
+def make_array() -> sc.DataArray:
+    da = sc.DataArray(make_var())
+    da.coords['xx'] = sc.arange('xx', 4, dtype='int64')
+    da.coords['yy'] = sc.arange('yy', 3, dtype='int64')
+    return da
+
+
+def make_dataset() -> sc.Dataset:
+    ds = sc.Dataset()
+    ds['xy'] = make_array()
+    ds['x'] = ds.coords['xx']
+    return ds
+
+
+@pytest.mark.parametrize("obj", [make_var(), make_array(), make_dataset()])
+@pytest.mark.parametrize("pos", [0, 1, 2, 3, -2, -3, -4])
+def test_length_1_list_gives_corresponding_length_1_slice(obj, pos):
+    assert sc.identical(obj['xx', [pos]], obj['xx', pos:pos + 1])
+
+
+@pytest.mark.parametrize("obj", [make_var(), make_array(), make_dataset()])
+def test_omitting_dim_when_slicing_2d_object_raises_DimensionError(obj):
+    with pytest.raises(sc.DimensionError):
+        obj[[3, 1]]
+
+
+@pytest.mark.parametrize("obj", [make_var(), make_array(), make_dataset()])
+def test_omitted_dim_is_equivalent_to_unique_dim(obj):
+    obj = obj['yy', 0]
+    assert sc.identical(obj[[3, 1]], obj[obj.dim, [3, 1]])
+
+
+@pytest.mark.parametrize("obj", [make_var(), make_array(), make_dataset()])
+def test_every_other_index_gives_stride_2_slice(obj):
+    assert sc.identical(obj['xx', [0, 2]], obj['xx', 0::2])
+    assert sc.identical(obj['xx', [1, 3]], obj['xx', 1::2])
+
+
+@pytest.mark.parametrize("obj", [make_var(), make_array(), make_dataset()])
+def test_unordered_indices_yields_result_with_reordered_slices(obj):
+    assert sc.identical(obj['xx', [2, 3, 0]],
+                        sc.concat([obj['xx', 2:4], obj['xx', 0:1]], 'xx'))
+
+
+@pytest.mark.parametrize("obj", [make_var(), make_array(), make_dataset()])
+def test_duplicate_indices_duplicate_slices_in_output(obj):
+    s1 = obj['xx', 1:2]
+    s2 = obj['xx', 2:3]
+    assert sc.identical(obj['xx', [2, 1, 1, 2]], sc.concat([s2, s1, s1, s2], 'xx'))
+
+
+@pytest.mark.parametrize("obj", [make_var(), make_array(), make_dataset()])
+def test_reversing_twice_gives_original(obj):
+    assert sc.identical(obj['xx', [3, 2, 1, 0]]['xx', [3, 2, 1, 0]], obj)
+
+
+@pytest.mark.skip("Inconsistent behavior with slice by mask but consistent with concat")
```

Comment:
Not sure how to address this inconsistency. Should we change the behavior of concat? Drop the edges manually here?

Reply:
I don't think slicing by index list should work with bin edges. It should probably raise an exception so that the user has to remove such a coord themselves.